fastjet/internal/delaunay/triangle.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2017 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package delaunay
+
+import (
+	"fmt"
+	"math"
+)
+
+// Triangle is a set of three points that make up a triangle, with hierarchical information to find triangles.
+type Triangle struct {
+	children []*Triangle // children are triangles that lead to the removal of this triangle
+	A, B, C  *Point
+}
+
+// Returns a triangle formed out of the three given points
+func NewTriangle(a, b, c *Point) *Triangle {
+	// make it counter clockwise
+	if clockwise(a, b, c) {
+		b, c = c, b
+	}
+	return &Triangle{
+		A: a,
+		B: b,
+		C: c,
+	}
+}
+
+func clockwise(a *Point, b *Point, c *Point) bool {
+	return (b.Y-a.Y)*(c.X-b.X)-(c.Y-b.Y)*(b.X-a.X) > 0
+}
+
+// inTriangle checks whether the point is in the triangle and whether it is on an edge
+// using barycentric coordinates
+func (t *Triangle) inTriangle(p *Point) (inside, edge bool) {
```

Comment:
This could benefit from inversion of the signature. It reads more naturally as a method on `*Point`, `func (p *Point) inTriangle(t *Triangle) (inside, edge bool)`.